source/mainnet/smart-contracts/tutorials/wCCD/wCCD-front-end-set-up.rst

```diff
@@ -4,6 +4,354 @@
+to your potential customers. Web front-ends are a familiar sight nowadays, but keep in mind
+that downloading a crypto wallet as a browser extension and its behavior or safe usage might be new
+for people using your front-end. Providing comprehensive explanations and step-by-step guides on your website on topics
+related to the browser wallet is important for a good user experience. The browser wallet
+connects via ``HTTPS`` to a server that is connected to a Concordium node. This setup alleviates the
+need for the user to host their own Concordium node.
+
+.. note::
+
+    Before you start with part 3 of this tutorial, make sure you have:
+
+    - access to the `Chrome <https://www.google.com/chrome/>`_ web browser on your laptop or computer
+
+    - `git <https://git-scm.com/>`_ installed to be able to clone a repository from `GitHub <https://github.com/>`_
+
+    - `node.js <https://nodejs.org/en/download/>`_ and `yarn <https://yarnpkg.com/getting-started/install>`_ installed
+
+    ``Node.js`` is a JavaScript runtime environment and ``yarn`` is a package manager
+    used in combination with ``node.js``. You can download `node.js <https://nodejs.org/en/download/>`_  and
+    `yarn <https://yarnpkg.com/getting-started/install>`_  on their official websites.
+
+Concordium browser wallet
+-------------------------
+
+A browser wallet is a piece of code that can be added as an extension to supported browsers such as ``Chrome``.
+The browser wallet hosts the private keys corresponding to the accounts of the user and a link that points
+to a `JSON-RPC server  <https://github.com/Concordium/concordium-json-rpc>`_.
+
+Your front-end code that is run in the browser constructs the transaction object
+and sends it the the browser wallet. The transaction object is signed by the private key hosted in the browser wallet
+and transmitted to the ``JSON-RPC server`` via ``HTTPS``. This server has access to a Concordium node and masks
+the request (including the signed transaction object) that comes via ``HTTPS`` from the browser wallet
+to a ``JSON-RPC`` request that the Concordium node can execute. The signed transaction is
+transmitted via peer-to-peer communication to other Concordium nodes and becomes
+part of the Concordium blockchain.
+
+.. warning ::
+
+    The Concordium browser wallet is in active development. A pre-MVP
+    (minimum viable product) version is currently available to be built from the source code.
+    Expect some breaking changes until the MVP version is released in the next quarter. It is not recommended to
+    use any pre-MVP product on mainnet.
+
+You can choose from two different workflows that will guide you through the setup steps. Workflow 1 is for
+advanced readers that want to build all the components from the source code and connect the browser wallet
+to their own hosted Concordium node. Workflow 2 is easier.
+The ``JSON-RPC wallet proxy`` is hosted by Concordium in workflow 2 which will take care of the
```

Comment:
Is that (``JSON-RPC wallet proxy``) how we usually render names? It looks a bit strange when rendered.

Reply:
yes, we render names in the documentation like this. The spacing between words seems slightly wide on my browser when using this name rendering style. There is a CSS file with some global style definitions. Since this affects the whole documentation, I left the global styles untouched (would need Danielle's feedback on it and checks how we should change style).